server/base/src/main/java/org/apache/accumulo/server/master/balancer/HostRegexTableLoadBalancer.java

```diff
@@ -244,11 +245,15 @@ protected void parseConfiguration(ServerConfiguration conf) {
             }
             String tableName = customProp.getKey().substring(HOST_BALANCER_PREFIX.length());
             String regex = customProp.getValue();
-            poolNameToRegexPattern.put(tableName, Pattern.compile(regex));
+            poolNameToRegexPatternBuilder.put(tableName, Pattern.compile(regex));
           }
         }
       }
     }
+
+    tableIdToTableName = ImmutableMap.copyOf(tableIdToTableNameBuilder);
```

Comment:
Not completely sure, but I am wondering if tableIdToTableName and poolNameToRegexPattern should be made volatile.  Before this change the reference to the map never changed.  Now the ref is being changed to a new map.  Making it volatile would ensure other threads see the new map.

Reply:
Good catch; I will update it.